### PR TITLE
Add kubectl tool

### PR DESCRIPTION
**Pull request by zregvart**

### CODE REVIEW

**Comment by zregvart** [comment]
We might consider something akin to this in lieu of #1918. cc @scoheb

**Comment by lcarva** [comment]
Nice!

File: Dockerfile [hunk 1]
```diff
@@ -32,6 +32,10 @@ WORKDIR /build
 COPY go.mod go.sum ./
 RUN go mod download
 
+# Copy the tools mod file for better layer caching when building locally
+COPY tools/go.mod tools/go.sum ./tools/
+RUN cd tools && go mod download
```
Comment (lcarva) [inline Dockerfile:37]: To make this happen we need an additional entry here: https://github.com/enterprise-contract/ec-cli/blob/d12edd162458b07ecbab38fb1ba3e08d248fc46f/.tekton/cli-main-ci-pull-request.yaml#L32-L33

Maybe we need another sub-module, or just add this to the one in `/`? Not sure if we want the SBOM of our image to include everything that is in under `tools`.

**Comment by scoheb** [comment]
looks good to me!

File: Dockerfile (hunk 1 above)
Comment (zregvart) [reply Dockerfile]: Opted to do a separate golang module. TIL: we already have `tools/go.mod` and `acceptance/go.mod` in the SBOM, see #1924

**Comment by zregvart** [comment]
Ready for review now...

File: Makefile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-MAKEFLAGS+=-j --no-print-directory
+MAKEFLAGS+=--no-print-directory
```
Comment (lcarva) [inline Makefile:1]: What's this for?

**Review by lcarva — APPROVED** [review]

**Pull request merged**